### PR TITLE
couchdb uses shared adapter main

### PR DESCRIPTION
**Pull request by lionelvillard**
## Proposed Changes

  * consolidate couchdb adapter main code.

### CODE REVIEW

File: couchdb/source/pkg/adapter/adapter.go [hunk 1]
```diff
@@ -18,68 +18,87 @@ package adapter
...
 		changeArgs: couchdb.ChangeArguments{},
 	}
-
-	return a, nil
 }
 
-func (a *adapter) Start(stopCh <-chan struct{}) error {
+func (a *couchDbAdapter) Start(stopCh <-chan struct{}) error {
 	// Just Poll for now. Other modes (long running, continues) will be provided later on.
 	return wait.PollUntil(2*time.Second, a.send, stopCh)
```
Comment (matzew) [inline couchdb/source/pkg/adapter/adapter.go]: why is that `PollUntil` ?

File: couchdb/source/pkg/adapter/adapter.go [hunk 2]
```diff
@@ -18,68 +18,87 @@ package adapter
...
+	rawusername, err := ioutil.ReadFile(env.CouchDbCredentialsPath + "/username")
+	if err != nil {
+		logger.Fatal("Missing username key in secret", zap.Error(err))
 	}
 
-	a := &adapter{
-		ce:         ceClient,
-		logger:     logger,
-		namespace:  options.Namespace,
+	rawpassword, err := ioutil.ReadFile(env.CouchDbCredentialsPath + "/password")
```
Comment (matzew) [inline couchdb/source/pkg/adapter/adapter.go]: Not for now, but open issue to support secrets ?

Comment (matzew) [reply couchdb/source/pkg/adapter/adapter.go]: Kafka has that - and we could see to share some ?

File: couchdb/source/pkg/adapter/adapter.go (hunk 1 above)
Comment (matzew) [reply couchdb/source/pkg/adapter/adapter.go]: Our PromQL does similar - I wonder if that generally can be improved   ( not saying lije CronJob .. .but ... :-) )

File: couchdb/source/pkg/adapter/adapter.go (hunk 2 above)
Comment (lionelvillard) [reply couchdb/source/pkg/adapter/adapter.go]: the adapter deployment mounts the secret.

**Comment by lionelvillard** [comment]
@vaikas-google can you approve?

**Pull request merged**